cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -89,12 +89,15 @@ private module ImplCommon {
     enclosing = arg.getEnclosingCallable()
   }
 
+  pragma[noinline]
+  private ParameterNode getAParameter(DataFlowCallable c) { result.getEnclosingCallable() = c }
```

Comment:
should this be a method on `DataFlowCallable`?

Reply:
I agree that would be nicer, but `DataFlowCallable` is language-specific, so that would mean duplicated code. If only QL had [extension predicates](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods)...